lib/functoria/lib.ml

```diff
@@ -47,32 +54,28 @@ module Config = struct
     in
     Key.Set.fold f all_keys skeys
 
-  let v ?(keys = []) ?(packages = []) ?(init = []) ~build_cmd ~src name main_dev
-      =
+  let v ?(keys = []) ?(packages = []) ?(init = []) ~build_cmd ~src name jobs =
     let name = Name.ocamlify name in
-    let jobs = Device_graph.create main_dev in
     let packages = Key.pure @@ packages in
+    let jobs = Impl.abstract jobs in
     let keys = Key.Set.(union (of_list keys) (get_if_context jobs)) in
     { packages; keys; name; init; jobs; build_cmd; src }
 
-  let eval ~partial context
+  let simplify ~partial context
```

Comment:
you probably don't need to rename that function

Reply:
Fair enough. I though the name was a bit misleading, since it doesn't call `eval`.